python/test/unit/conftest.py

```diff
@@ -37,6 +37,8 @@ def fresh_config(request):
     try:
         for name, conf in config_map.items():
             setattr(config, name, conf.copy().reset())
+            for knob in conf.knob_descriptors.values():
+                monkeypatch.delenv(knob.key, raising=False)
```

Comment:
Isn't this redundant because you use `monkeypatch.setenv` which should restore the prior state when the test is finalized?

Reply:
For restoring environment yes, but the issue I'm targeting is any initial assert against default values, e.g. https://github.com/triton-lang/triton/blob/main/python/test/unit/test_config.py#L108 . For that specific assert if you set `TRITON_DEBUG=1` in the environment before running the test then the test will fail.